src/stactools/sentinel3/file_extension_updated.py

```diff
@@ -0,0 +1,40 @@
+from typing import List, Optional
+
+import pystac
+from pystac.extensions.file import ByteOrder, FileExtension, MappingObject
+
+LOCAL_PATH_PROP = "file:local_path"
+
+
+class FileExtensionUpdated(FileExtension):
+    SCHEMA_URI = "https://stac-extensions.github.io/file/v2.1.0/schema.json"
+
+    def apply(self,
+              byte_order: Optional[ByteOrder] = None,
+              checksum: Optional[str] = None,
+              header_size: Optional[int] = None,
+              size: Optional[int] = None,
+              values: Optional[List[MappingObject]] = None,
+              local_path: Optional[str] = None) -> None:
+        super().apply(byte_order, checksum, header_size, size, values)
+        self.local_path = local_path
+
+    @property
+    def local_path(self) -> Optional[str]:
+        """Get or sets the local path of the file."""
+        return self._get_property(LOCAL_PATH_PROP, str)
```

Comment:
I'm not sure what the value of `local_path` is here. Most downstream users of these data will be hosting these STAC items over the network, so `local_path` would end up being pretty meaningless, I think?

Reply:
This is a property to reflect a relative path that can be used by clients for different purposes to organize the files locally, which is taken from [file info extension spec](https://github.com/stac-extensions/file#local-path).

Reply:
Can you give a use-case? I'm unclear how `local_path` would be used.

Reply:
So, a relative path taken from product dir to individual files can be used If someone decides to copy the product files somewhere else, the local path would still point to the files from the project dir, whereas the full URL path would be invalid and the path would need to be parsed from it separately.

Another use case would be that [SNAP toolbox](https://step.esa.int/main/) reads the SAFE manifest to open a product that references the different files (e.g. image bands) relatively and thus expects to find them in the referenced folder (e.g. IMG_DATA).

Reply:
What is the `local_path` relative to? The item's self href? If so, how it it different than a relative `href`?

Your SAFE manifest example feels more like a "subasset" structure, where one asset references another asset -- but maybe I'm just not understanding the problem space correctly.

Do you have any open examples of tooling that uses `local_path`? That might help me understand better. I know **pystac** doesn't use it at all.

Reply:
The stac items for Sentinel-3 on [Planetary Computer](https://planetarycomputer.microsoft.com/dataset/storage/sentinel-3) use this property. They have something like below:
```
"href" : "https://sentinel3euwest.blob.core.windows.net/sentinel-3/SLSTR/SL_2_FRP___/2023/03/10/S3B_SL_2_FRP____20230310T235936_20230311T000236_20230312T001332_0179_077_087_4500_PS2_O_NT_004.SEN3/cartesian_in.nc"
```
```
"file:local_path" : "S3B_SL_2_FRP____20230310T235936_20230311T000236_20230312T001332_0179_077_087_4500_PS2_O_NT_004.SEN3/cartesian_in.nc"
```

Reply:
Okay, after re-(and re-) reading the description in [the extension](https://github.com/stac-extensions/file#local-path) I guess I can see what it's used for. Feels pretty fragile to require files to be laid out a certain way on a filesystem but I guess that's how things are 🤷🏽.

@chorng thanks for your patience iterating this w/ me.